MonkeyWrench.Web.UI/EditLane.aspx.cs

```diff
@@ -705,7 +706,16 @@ protected void cmdSave_Click (object sender, EventArgs e)
 		lane.enabled = chkEnabled.Checked;
 		lane.additional_roles = txtRoles.Text;
 		lane.priority = Int32.Parse (lstPriority.SelectedItem.Value);
-		lane.is_protected = chkProtected.Checked;
+		lane.is_protected = chkProtected.Checked;
+
+		// Don't ever pull more than 100 commits
+		try {
+			int commits = Int32.Parse (txtFetchCommits.Text);
+			lane.max_commits_to_fetch = commits >= 100 ? 100 : commits;

```

Comment:
This would be a hard limit the user couldn't get around, so I don't agree here.

We're already setting the default to 1 (including for all existing lanes), so this would only happen if someone accidentally putting in some high number in the text box. Given that the user can recover from some accidental high number (he can mark all as "don't build", or just delete every revision), I'd opt for not having any hard limits.